nest/src/users/users.controller.ts

```diff
@@ -114,13 +101,32 @@ export class UsersController {
       },
     },
   })
-  @UseInterceptors(FileInterceptor('file', storage))
-  uploadFile(
-    @Param('name') name: string,
-    @UploadedFile() file: any,
-  ): Observable<unknown> {
-    console.log('name ' + name);
-    this.usersService.setFilename(file.filename, name);
+  @UseInterceptors(
+    FileInterceptor('file', {
+      storage: diskStorage({
+        destination: './uploads/profileimages',
+        filename: editFileName,
+      }),
+      fileFilter: imageFileFilter,
+    }),
+  )
```

Comment:
maybe add here MIME type validation ?
smth like :
```
  @ApiConsumes('multipart/form-data')
  @ApiBody({
    description: 'Profile image upload',
    type: 'multipart/form-data',
  })
```
also you can consider to add :
`@UseInterceptors(RateLimiterInterceptor)`
to prevent dos attacks , it s defined in the nestjs-rate-limiter lib

Reply:
I couldn't implemented the piece of code you wrote here, because otherwise we can't upload pictures from swagger http://localhost:8080/api